metadata_layer.py

```diff
@@ -646,7 +646,9 @@ def mirror_link(self, model_object, data_source, link, link_obj, policy):
                 extension=extension
             )
         else:
-            filename = representation.default_filename(link_obj)
+            filename = representation.default_filename(
+                link_obj, representation.media_type
+            )
```

Comment:
Without this, images mirrored from Content Cafe will have the ".aspx" extension even though that's obviously wrong. Passing in the known media type helps us guess better.